src/data_c.c4

```diff
@@ -628,6 +630,20 @@ SHMEM_DEFINE_FOR_RMA(`SHMEM_DEF_PUT_SIGNAL_NBI')
 SHMEM_DEFINE_FOR_SIZES(`SHMEM_DEF_PUT_N_SIGNAL_NBI')
 SHMEM_DEF_PUT_N_SIGNAL_NBI(`mem', `1')
 
+uint64_t SHMEM_FUNCTION_ATTRIBUTES
+shmemx_signal_fetch(const uint64_t* sig_addr)
```

Comment:
Woah, did we miss a context argument on this function?

Reply:
Yeah, I don't see why it wouldn't have one... 😕

Reply:
@jdinan Thanks. Yeah, I am confused about whether it needs a context argument as it is only reading a local memory. Then again, it needs to keep atomicity guarantees. In that perspective, it needs a context argument. If it is only a simple load operation, then I am not sure why we need to have an extra API. I wonder why we did not see these through during the proposal discussion.

Reply:
It makes sense to me to implement as an atomic fetch, since it can be overlapped with signal atomic updates.